File: split/alterboxdimen.pas
procedure alterboxdimen;
var
  c: smallnumber;
  b: eightbits;
begin
  c := curchr;
  scaneightbitint;
  b := curval;
  scanoptionalequals;
  scandimen(false, false, false);
  if eqtb[3678 + b].hh.rh <> 0 then
    mem[eqtb[3678 + b].hh.rh + c].int := curval;
end;

File: cpp/src/alterboxdimen.cpp
#include "alterboxdimen.h"
#include "lecture.h"

void alterboxdimen(Token t)
{
	auto c = t.chr;
	auto b = scaneightbitint();
	scanoptionalequals();
	int val = scan_normal_dimen();
	if (box(b))
		mem[box(b)+c].int_ = val;
}
